pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -79,17 +84,21 @@ func (d *DualWriterMode2) Get(ctx context.Context, name string, options *metav1.
 // List overrides the behavior of the generic DualWriter.
 // It returns Storage entries if possible and falls back to LegacyStorage entries if not.
 func (d *DualWriterMode2) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	ctx = klog.NewContext(ctx, d.Log)
+	log := d.Log.WithValues("kind", options.Kind, "resourceVersion", options.ResourceVersion)
```

Comment:
not sure if this was intentional, but did you want to make these values available in the new context logger? if so, should the order be changed here?
```suggestion
	log := d.Log.WithValues("kind", options.Kind, "resourceVersion", options.ResourceVersion)
	ctx = klog.NewContext(ctx, log)
```

Reply:
yes! And I'm making this mistake everywhere. Fixing it!